src/ILLink.RoslynAnalyzer/TrimAnalysis/TrimAnalysisVisitor.cs

```diff
@@ -118,15 +132,22 @@ public override void HandleArgument (MultiValue argumentValue, IArgumentOperatio
 			));
 		}
 
-		public override void HandleReceiverArgument (MultiValue receieverValue, IInvocationOperation operation)
+		// Similar to HandleArgument, for an assignment operation that is really passing an argument to a property setter.
+		public override void HandlePropertySetterArgument (MultiValue value, IMethodSymbol setMethod, ISimpleAssignmentOperation operation)
 		{
-			if (operation.Instance == null)
-				return;
+			var parameter = new MultiValue (new SymbolValue (setMethod.Parameters[0]));
 
-			MultiValue implicitReceiverParameter = new MultiValue (new SymbolValue (operation.TargetMethod, isMethodReturn: false));
+			TrimAnalysisPatterns.Add (new TrimAnalysisPattern (value, parameter, operation));
+		}
+
+		// Can be called for an invocation or a propertyreference
+		// where the receiver is not null (so an instance method/property).
+		public override void HandleReceiverArgument (MultiValue receiverValue, IMethodSymbol targetMethod, IOperation operation)
+		{
+			MultiValue implicitReceiverParameter = new MultiValue (new SymbolValue (targetMethod!, isMethodReturn: false));
 
 			TrimAnalysisPatterns.Add (new TrimAnalysisPattern (
-				receieverValue,
+				receiverValue,
 				implicitReceiverParameter,
 				operation
```

Comment:
I assume this is temporary - it doesn't make much sense to wrap the method in a MultiValue and pass it as something called `implicitReceiverParameter`. The value of the implicit receiver parameter I think is the `this`, which doesn't "match" the method. That said the pattern should record which method is being called...

Reply:
I renamed it to `thisParameter` - does that make more sense? The intention is to create a SingleValue representing the this parameter of the method. It's wrapped in a MultiValue because the pattern cache stores source/target as MultiValue, even though they will only contain single values in a lot of cases.

Reply:
I think for now this is fine. Going forward I think the plan was to change these classes to match the various shared operations and thus this code will change anyway.